### PR TITLE
gce: support assign interface of local ssd

### PR DESCRIPTION
**Pull request by amoskong**
GCE has two kind of interface for local ssd, NVME and SCSI.
We always use NVME in current code, this patch added support
to assign interface by `gce_local_ssd_if_db/loader/monitor`.

We will still use NVME by default.

### CODE REVIEW

**Comment by amoskong** [comment]
I will rebase it.

File: sdcm/cluster_gce.py [hunk 1]
```diff
@@ -181,6 +182,8 @@ def _get_root_disk_struct(self, name, disk_type='pd-standard', dc_idx=0):
 
     def _get_local_ssd_disk_struct(self, name, index, interface='NVME', dc_idx=0):
         device_name = '%s-data-local-ssd-%s' % (name, index)
+        if local_ssd_if is None:
+            local_ssd_if = 'NVME'
```
Comment (larisau) [inline sdcm/cluster_gce.py]: the variable local_ssd_if is not used

File: sdcm/cluster_gce.py [hunk 2]
```diff
@@ -127,7 +127,7 @@ class GCECluster(cluster.BaseCluster):
     def __init__(self, gce_image, gce_image_type, gce_image_size, gce_network, services, credentials,
                  cluster_uuid=None, gce_instance_type='n1-standard-1', gce_region_names=['us-east1-b'],
                  gce_n_local_ssd=1, gce_image_username='root', cluster_prefix='cluster',
-                 node_prefix='node', n_nodes=[10], add_disks=None, params=None):
+                 node_prefix='node', n_nodes=[10], add_disks=None, params=None, gce_local_ssd_if=None):
```
Comment (larisau) [inline sdcm/cluster_gce.py]: I think it's better to set default value as 'NVME':
`gce_local_ssd_if='NVME'`

**Comment by amoskong** [comment]
I have updated the PR to set default gce_local_ssd_if to None, and
removed `local_ssd_if`
usage inside _get_local_ssd_disk_struct(). Thanks Larisa.

**Comment by roydahan** [comment]
@amoskong, I don't understand why we need this PR...
What functionality does it add or what does it fix?

IIUC, before this PR, if the yaml had "0" for n_local_ssd" and has gce_pd_ssd_disk_size_db or gce_pd_standard_disk_size_db the behaviour will be the same.

Although we have "node.remoter.run('ls /dev/nvme0n*')" if we don't have any local_ssd (n_local_ssd: 0), it won't fail.

Am I wrong?

**Comment by amoskong** [comment]
Local ssd have two kind of interface: NVMe or SCSI.

When we use SCSI, the device file name is /dev/sd*, "node.remoter.run('ls
/dev/nvme0n*')" should not fail.

**Comment by larisau** [comment]
@amoskong The test working with pd_standard doesn't fail on any of 'ls' commands, you can see the device /dev/sdb, and there is no interface choice for it, it's always SCSI.
It fails on scylla_setup.

**Comment by roydahan** [comment]
So, this PR is to support local_ssd with SCSI interface?

**Comment by roydahan** [comment]
@amoskong, there are some conflicts.
However, I'm not sure we really need this PR.
if we need SCSI disk we can use pd-ssd or pd-standard, I don't see the use case of using local_ssd with SCSI interface.

If you think there is a use case for it, fix the conflicts and merge.

**Comment by amoskong** [comment]
I don't have usage for this new interface, let's close this PR.

**Pull request closed without merging**